diff --git a/crypto/src/lib.rs b/crypto/src/lib.rs
index 3f7ac38..eabeda9 100644
--- a/crypto/src/lib.rs
+++ b/crypto/src/lib.rs
@@ -43,7 +43,7 @@ pub struct MlsCiphersuite(openmls::prelude::Ciphersuite);
 
 impl Default for MlsCiphersuite {
     fn default() -> Self {
-        Self(openmls::prelude::Ciphersuite::MLS_128_DHKEMX25519_CHACHA20POLY1305_SHA256_Ed25519)
+        Self(openmls::prelude::Ciphersuite::MLS_128_DHKEMX25519_AES128GCM_SHA256_Ed25519)
     }
 }
 
